incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java

```diff
@@ -0,0 +1,142 @@
+import fr.openent.incidents.service.IncidentsTypeService;
+import fr.openent.presences.core.constants.Field;
+import fr.openent.presences.db.DB;
+import fr.openent.presences.db.DBService;
+import fr.wseduc.webutils.Either;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.entcore.common.sql.Sql;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+import org.powermock.reflect.Whitebox;
+
+import java.util.Collections;
+
+@RunWith(VertxUnitRunner.class)
+public class IncidentsTypeServiceTest extends DBService {
+
+    private Vertx vertx;
+
+    private IncidentsTypeService incidentsService;
+
+    @Before
+    public void setUp() {
+        vertx = Vertx.vertx();
+        Sql.getInstance().init(vertx.eventBus(), "fr.openent.incidents");
+        this.incidentsService = new DefaultIncidentsTypeService();
+    }
+
+    @Test
+    public void testFetchIncidentsType(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "SELECT * FROM null.incident_type where structure_id = ?";
+        JsonArray expectedParams = new JsonArray().add("structureId");
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "fetchIncidentsType", "structureId", (Handler) e -> {
+
+        });
+    }
+
+    @Test
+    public void testFetchUsedIncidentsType(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "WITH ids AS (SELECT i.id, i.label FROM null.incident_type i WHERE structure_id = ?) " +
+                "SELECT DISTINCT i.id, i.label FROM ids i WHERE (i.id IN (SELECT type_id FROM null.incident))";
+        JsonArray expectedParams = new JsonArray().add("structureId");
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", (Handler) e -> {
+
```

Comment:
Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", null)